docker/vllm_metax.Dockerfile

```
@@ -1,164 +1,174 @@
-FROM registry.access.redhat.com/ubi9/ubi:9.6
+ARG BUILD_BASE_IMAGE=registry.access.redhat.com/ubi9/ubi:9.6
+ARG PYTHON_VERSION=3.10
+ARG PIP_INDEX_URL
+ARG PIP_EXTRA_INDEX_URL=https://repos.metax-tech.com/r/maca-pypi/simple
+ARG UV_TRUSTED_HOST=repos.metax-tech.com
+ARG UV_INDEX_URL=${PIP_INDEX_URL}
+ARG UV_EXTRA_INDEX_URL=${PIP_EXTRA_INDEX_URL}
+ARG CU_BRIDGE_VERSION=3.1.0
+
+#################### BASE BUILD IMAGE ####################
+FROM ${BUILD_BASE_IMAGE} AS base
+
+ARG PYTHON_VERSION
+ARG PIP_INDEX_URL UV_INDEX_URL
+ARG PIP_EXTRA_INDEX_URL UV_EXTRA_INDEX_URL
+ARG UV_INDEX_URL=${PIP_INDEX_URL}
+ARG UV_EXTRA_INDEX_URL=${PIP_EXTRA_INDEX_URL}
+ARG UV_TRUSTED_HOST
+
+# Install system dependencies and uv, then create Python virtual environment
+RUN curl -LsSf https://astral.sh/uv/install.sh | sh && \
+    $HOME/.local/bin/uv venv /opt/venv --python ${PYTHON_VERSION} && \
+    rm -f /usr/bin/python3 /usr/bin/python3-config /usr/bin/pip && \
+    ln -s /opt/venv/bin/python3 /usr/bin/python3 && \
+    ln -s /opt/venv/bin/python3-config /usr/bin/python3-config && \
+    ln -s /opt/venv/bin/pip /usr/bin/pip && \
+    python3 --version && python3 -m pip --version
```

Note: this is going to download install an astral-provided python version, it'd be better to bump python to 3.12 and use the `dnf`-provided version:

```suggestion
ENV VIRTUAL_ENV=/opt/venv
ENV PATH=${VIRTUAL_ENV}/bin:$PATH
RUN dnf install python${PYTHON_VERSION}-pip && \
    dnf clean all && \
    python${PYTHON_VERSION} -m pip install --no-cache uv && \
    uv venv --python=${PYTHON_VERSION} && \
    python --version && \
    uv self version
```